### DISCUSSION TITLE
set bg color in matplotlib figure

### DISCUSSION THREAD

**Discussion by bigfooted** [Kitware/trame#317] [Q&A] [answered]
Hi! In a matplotlib figure, I would like to set the figure background color, so the color of the figure outside of the plotting region: https://matplotlib.org/stable/api/figure_api.html
This should be done by setting the facecolor of the figure, but it does not work. I can only set the facecolor of the axes.

So it does not work like the example of the website:
```python
fig, ax = plt.subplots(figsize=(2, 2), facecolor='lightskyblue',
                       layout='constrained')
```
also not like this:

```python
mpl.rcParams["figure.facecolor"]="black"
```

I guess it is overruled by some trame layout rule? Is there a way to get this to work in a matplotlib way, or I have to do this elsewhere?

**Comment by jourdain** [comment]
I'm not sure, but that could be a limitation of the library we rely on to map the matplotlib definition from the server to the client via D3.

We are using [mpld3](https://github.com/Kitware/trame-matplotlib/blob/master/vue-components/package.json#L13) and the [website is here](https://mpld3.github.io/).

But looking at the version, they seem to have provided an updated one 9 months ago, while ours is 2 years old. It might be worth trying to bump that version and see what happens. 

Are you up for trying and testing it yourself? 

The gist would be

```
git clone https://github.com/Kitware/trame-matplotlib.git
cd trame-matplotlib/vue-components

# edit package.json to bump the mpld3 version to "0.5.9"

# Build the web part
npm install
npm run build
pip install ..

# Test your app with the not released trame-matplotlib...
```

**Comment by jourdain** [comment]
Another option is to set the CSS background style on the div containing the plot. (maybe)

**Comment by bigfooted** [comment, marked as answer]
Thanks for the hints, I upgraded mpl3d but that did not work. However, you can change the background from tramematplotlib.Figure,

`                    html_figure = tramematplotlib.Figure(style="position: absolute",classes="grey ma-0 pa-0")
`
